text/0000-katello-as-a-plugin.md

```diff
@@ -0,0 +1,107 @@
+
+#### Design
+
+This has been previously discussed and outlined by an existing [foreman-dev post](https://groups.google.com/forum/#!topic/foreman-dev/hebniL0yyqM). The larger proposals would be:
+
+ * Add Katello plugin manifest to puppet-foreman
+ * Add Katello plugin manifests for each backend service to puppet-foreman
+ * Add puppet-capsule features to puppet-foreman_proxy
+
+#### Drawbacks
+
+Adds a lot more manifests to puppet-foreman and puppet-foreman_proxy and keeps things less separated and focused.
+
+## Installer Repositories
+
+The `katello-installer` uses the foreman-installer and scenarios in order to install providing it's own scenario for Katello and for the Capsule. These including the system checks, and hooks are all contained within the `katello-installer` repository.
+
+### Design
+
+Unify the installer by storing multiple scenarios and dependencies in the single `foreman-installer` repository. This would put everything in one place and encourage more design work to support a variety of scenarios for different deployment types.
+
+### Drawbacks
+
+Merging the two together could muddy the waters with respect to different sets of hooks, requirements and modules required to install a given scenario. 
+
+## Release Repositories
+
+Currently the Foreman repositories are housed at [http://yum,theforeman.org](http://yum.theforeman.org) and Katello's at [https://fedorapeople.org/groups/katello/releases/yum/](https://fedorapeople.org/groups/katello/releases/yum/).
+
+#### Design
+
+Move the Katello repositories to be served from the Foreman repository location.
+
+#### Drawbacks
+
+None.
+
+## Documentation
+
+Today the Foreman website and plugins are kept in a different repository than Katello. Further, the websites and documentation are at different URLs [http://www.theforeman.org](http://theforeman.org) and [http://www.katello.org](http://www.katello.org).
+
+#### Design
+
+Move Katello's documentation inside the plugin section of the Foreman's website documentation and get rid of the katello.org repository and website.
+
+#### Drawbacks
+
+The effort to re-do the website design, and include all of the current information displayed such as troubleshooting, user guides, and developer information.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The grand drawback with this work would be that it would interrupt business as usual and no changes or work is needed that would disrupt the release flow of Katello.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+ * What to do about nested Organizations given Katello does not know how to support this given it's organization model.
```

Comment:
I tend to agree and was planning to open some additional RFCs as these items get more to functionality and/or changing the architecture of Katello which I think is made easier by first making Katello a regular feeling plugin.